src/odemis/driver/test/technolution_test.py

```diff
@@ -1420,6 +1454,47 @@ def test_get_field_and_mega_field_combination(self):
         self.assertEqual(field_images[0] * field_images[1], self.counter)
         self.assertEqual(field_images[0] * field_images[1], self.counter2)
 
+class Test_AsmApiException(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        if TEST_NOHW:
+            raise unittest.SkipTest('No simulator for the ASM and HwComponents present. Skipping tests.')
+
+    @classmethod
+    def tearDownClass(cls):
+        pass
+
+    def setUp(self):
+        self.ASM_manager = AcquisitionServer("ASM", "main", URL, children=CHILDREN_ASM, externalStorage=EXTRNAL_STORAGE)
+        for child in self.ASM_manager.children.value:
+            if child.name == CONFIG_MPPC["name"]:
+                self.MPPC = child
+            elif child.name == CONFIG_SCANNER["name"]:
+                self.EBeamScanner = child
+            elif child.name == CONFIG_DESCANNER["name"]:
+                self.MirrorDescanner = child
+
+        # Ensure that only empty images will be received
+        self.MPPC.dataContent.value = "empty"
+
+        resp = self.ASM_manager._session.get(self.ASM_manager._host + "/scan/clock_frequency",
+                                             json=None, timeout=600)
+
+        self.error_object = AsmApiException("test_url", resp, 404, self.ASM_manager)
+
+    def tearDown(self):
+        self.ASM_manager.terminate()
+        time.sleep(0.2)
+
+    def test_AsmApiException(self):
+        with self.assertRaises(Exception):
+            raise self.error_object
+
+    def test_system_checks(self):
+        # This method which is tested is mainly dependent on making external calls so the test is to see if the method
+        # works.
+        self.error_object.system_checks(self.ASM_manager)
```

Comment:
See my comments on the implementation for the system checks. Might want to add a test to check the queue filling rate.